src/treetensornetworks/projttns/projttnsum.jl

```diff
@@ -1,17 +1,25 @@
 """
 ProjTTNSum
 """
-struct ProjTTNSum{V}
-  pm::Vector{ProjTTN{V}}
-  function ProjTTNSum(pm::Vector{ProjTTN{V}}) where {V}
-    return new{V}(pm)
+struct ProjTTNSum{T<:AbstractProjTTN}
+  pm::Vector{T} where {T}
+  function ProjTTNSum(pm::Vector{T}) where {T}
+    return new{T}(pm)
   end
 end
+#ToDo: define accessor functions
 
 copy(P::ProjTTNSum) = ProjTTNSum(copy.(P.pm))
 
+# The following constructors don't generalize well, maybe require to pass AbstractProjTTN instead of TTN and remove these? 
 ProjTTNSum(ttnos::Vector{<:TTN}) = ProjTTNSum([ProjTTN(M) for M in ttnos])
+function ProjTTNSum(init_states::Vector{<:TTN}, ttnos::Vector{<:TTN})
+  return ProjTTNSum([
+    ProjTTNApply(state, operator) for (state, operator) in zip(init_states, ttnos)
+  ])
+end
 
+# This constructor can't differentiate between different concrete AbstractProjTTN, remove?
 ProjTTNSum(Ms::TTN{V}...) where {V} = ProjTTNSum([Ms...])
```

Comment:
```suggestion
ProjTTNSum(operators::AbstractTTN...) = ProjTTNSum([operators...])
```

Reply:
Though I would actually prefer to get rid of this constructor, and only have constructors that expect `Vector{<:AbstractTTN}` or `Vector{<:AbstractProjTTN}`.

Reply:
I agree. Do you think it's enough to just have a constructor for `Vector{<:AbstractProjTTN}`? The other one would just default to making `ProjTTN`s, which is also ok.

Reply:
I think keeping a definition `ProjTTNSum(operators::Vector{<:AbstractTTN}) = ProjTTNSum(ProjTTN.(operators))` makes sense to me, since (at least to me) it is pretty obvious what it should return.